### PR TITLE
Added support for PostNord

### PR DESCRIPTION
**Pull request by JaisDK**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,6 +29,7 @@ before_install:
 
     - cp tests/Application/.env.test.dist tests/Application/.env.test
     - set -a && source tests/Application/.env.test && set +a
+    - composer config -g github-oauth.github.com "$TOKEN"
```
Comment (loevgaard) [inline .travis.yml]: Why this?

File: composer.json [hunk 2]
```diff
@@ -8,7 +8,8 @@
         "php": "^7.2",
         "ext-soap": "*",
 
-        "sylius/sylius": "~1.3.0"
+        "sylius/sylius": "~1.3.0",
+        "lsv/pd-parcelshop-api": "dev-master"
```
Comment (loevgaard) [inline composer.json]: `dev-master` is not an option. It needs to be a stable release.

File: spec/Manager/ProviderManagerSpec.php [hunk 3]
```diff
@@ -12,11 +12,14 @@
 
 final class ProviderManagerSpec extends ObjectBehavior
```
Comment (loevgaard) [inline spec/Manager/ProviderManagerSpec.php]: Why change this file at all? It's the spec for the `ProviderManager`

File: src/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('setono_sylius_pickup_point');
+        $rootNode
+            ->addDefaultsIfNotSet()
+            ->children()
+                ->arrayNode('postnord')
```
Comment (loevgaard) [inline src/DependencyInjection/Configuration.php]:
```suggestion
                ->arrayNode('post_nord')
```

This goes for all the other places using `postnord`

File: src/Manager/ProviderManager.php [hunk 5]
```diff
@@ -41,6 +48,11 @@ public function all(): array
         return $this->providers;
     }
 
+    /**
+     * @param string $code
+     *
+     * @return bool
+     */
```
Comment (loevgaard) [inline src/Manager/ProviderManager.php]: This phpdoc belongs in the interface

File: src/Manager/ProviderManager.php [hunk 6]
```diff
@@ -14,6 +14,11 @@ class ProviderManager implements ProviderManagerInterface
      */
     private $providers = [];
 
+    /**
+     * ProviderManager constructor.
+     *
+     * @param ProviderInterface ...$providers
+     */
```
Comment (loevgaard) [inline src/Manager/ProviderManager.php]: Remove this phpdoc

File: src/Provider/PostNordProvider.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+use Lsv\PdDk\Exceptions\MalformedAddressException;
+use Lsv\PdDk\Exceptions\ParcelNotFoundException;
+use Setono\SyliusPickupPointPlugin\Model\PickupPoint;
+use Setono\SyliusPickupPointPlugin\Model\PickupPointInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+final class PostNordProvider implements ProviderInterface
+{
+    use ContainerAwareTrait;
```
Comment (loevgaard) [inline src/Provider/PostNordProvider.php]: Inject dependencies instead

File: .travis.yml (hunk 1 above)
Comment (JaisDK) [reply .travis.yml]: This will be removed once I'm done running tests in my fork

File: composer.json (hunk 2 above)
Comment (JaisDK) [reply composer.json]: There's no stable, I'll write the owner to make one

File: spec/Manager/ProviderManagerSpec.php (hunk 3 above)
Comment (JaisDK) [reply spec/Manager/ProviderManagerSpec.php]: isEnabled needed to be set

File: src/Manager/ProviderManager.php (hunk 6 above)
Comment (JaisDK) [reply src/Manager/ProviderManager.php]: why no doc?

File: src/Manager/ProviderManager.php (hunk 5 above)
Comment (JaisDK) [reply src/Manager/ProviderManager.php]: The whole method is missing from the interface. I'll add :)

File: src/Provider/PostNordProvider.php (hunk 7 above)
Comment (JaisDK) [reply src/Provider/PostNordProvider.php]: How is that better that this trait?

Comment (loevgaard) [reply src/Provider/PostNordProvider.php]: You should inject the things you need from the container in the class instead of just injecting the container.

Comment (JaisDK) [reply src/Provider/PostNordProvider.php]: How would I do the parameter check on line 130?

File: src/Provider/PostNordProvider.php [hunk 8]
```diff
@@ -127,6 +144,6 @@ public function getName(): string
      */
     public function isEnabled(): bool
     {
-        return $this->container->hasParameter('setono_sylius_pickup_point_post_nord_apikey');
+        return !is_null($this->apiKey);
```
Comment (loevgaard) [inline src/Provider/PostNordProvider.php]: `return null !== $this->apiKey;`

**Pull request merged**